caas/kubernetes/provider/specs/admissionregistration_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package specs_test
+
+import (
+	"encoding/base64"
+	"strings"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sspecs "github.com/juju/juju/caas/kubernetes/provider/specs"
+	"github.com/juju/juju/testing"
+)
+
+type webhooksSuite struct {
+	testing.BaseSuite
+}
+
+var _ = gc.Suite(&webhooksSuite{})
+
+func (s *webhooksSuite) TestK8sMutatingWebhookV1Beta1(c *gc.C) {
```

Comment:
Is there a v1 test also?

Reply:
Even the code has added support for both v1 and v1beta1 for webhooks but it's currently not able to detect v1 because the v1 and v1beta1 have the exact same struct currently (but just some certain fields expect different default values).
To have better compatibility, we try v1beta1 before v1. That means any validate webhooks spec will be parsed as v1beta1 always unless v1 introduces a diff in the struct in the future.

Reply:
Well lets add a test to make sure the v1 path is never taken. That way when a change to v1 that disambiguates v1 from v1beta we find out.